src/classes/CW_AudienceGroupControllerTest.cls

```diff
@@ -0,0 +1,333 @@
+        insert camp;
+        
+        List<CW_Audience_Group__c> listGroups = TestDataUtil.createAudienceGroups(camp.Id,5);
+        listGroups[0].Is_Control_Group__c = false;
+        listGroups[0].Control_Group__c = listGroups[1].Id;
+        listGroups[2].Is_Control_Group__c = false;
+        listGroups[2].Control_Group__c = listGroups[1].Id;
+        insert listGroups;
+        
+        User thisUser = [SELECT Id FROM User WHERE Id = :UserInfo.getUserId()];
+    	
+    	Test.startTest();
+    	
+		System.runAs(thisUser)
+	    {  
+	        Test.setCurrentPage(Page.CW_Campaign_Information);
+			CW_WizardController wizCtrl = new CW_WizardController(new ApexPages.StandardController(camp));
+			wizCtrl.next();
+			
+			Test.setCurrentPage(Page.CW_Audience);
+			Apexpages.currentPage().getParameters().put('id',camp.Id);
+			CW_AudienceGroupController agCtrl = new CW_AudienceGroupController();
+			
+			agCtrl.editGroups();
+			
+			agCtrl.getAudienceGroup();
+			
+			agCtrl.campGroups[0].aGroup.Is_Control_Group__c = true;
+			agCtrl.saveGroups();
+			
+			listGroups = [SELECT Id,Name,Description__c,
+								 Is_Control_Group__c,
+								 Control_Group__c
+						  FROM CW_Audience_Group__c
+						  WHERE Campaign__c = : camp.Id];
+						  
+			system.assertEquals(listGroups[0].Control_Group__c,null);
+			system.assertEquals(listGroups[0].Is_Control_Group__c,true);
+			
+	    }
+	    Test.stopTest();
+    }
+	
+	
+    static testMethod void deleteGroupsTest() 
+    {
+        initCustomSettings();
+        
+        CW_Campaign__c camp = TestDataUtil.createCampaign();
+        insert camp;
+        
+        List<CW_Audience_Group__c> listGroups = TestDataUtil.createAudienceGroups(camp.Id,5);
+        insert listGroups;
+        
+        User thisUser = [SELECT Id FROM User WHERE Id = :UserInfo.getUserId()];
+    	
+    	Test.startTest();
+    	
+		System.runAs(thisUser)
+	    {  
+	        Test.setCurrentPage(Page.CW_Campaign_Information);
+			CW_WizardController wizCtrl = new CW_WizardController(new ApexPages.StandardController(camp));
+			wizCtrl.next();
+			
+			Test.setCurrentPage(Page.CW_Audience);
+			Apexpages.currentPage().getParameters().put('id',camp.Id);
+			CW_AudienceGroupController agCtrl = new CW_AudienceGroupController();
+			
+			system.assertEquals(agCtrl.getPageNumber(),1);
+			system.assertEquals(agCtrl.setGroups.getResultSize(),listGroups.size());
+			
+			agCtrl.getAudienceGroup();
+			agCtrl.contextItem = agCtrl.campGroups[0].aGroup.Id;
+			agCtrl.doSelectItem();
+			agCtrl.contextItem = agCtrl.campGroups[1].aGroup.Id;
+			agCtrl.doSelectItem();
+			agCtrl.deleteGroups();
+			
+			listGroups = [SELECT Id,Name,Description__c
+						  FROM CW_Audience_Group__c
+						  WHERE Campaign__c = : camp.Id];
+						  
+			system.assertEquals(agCtrl.setGroups.getResultSize(),listGroups.size());
+			
+	    }
+	    Test.stopTest();
+    }
+    
+    static testMethod void createNewGroup()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility